### ISSUE TITLE
Some problems in tool.py

### ISSUE THREAD

**Issue by multimedia-infomation-security** [DTaoo/Discriminative-Sounding-Objects-Localization#13]
I don't know how to use the function‘visualize’ in the tool.py as follows:
def visualize(images, cams, e):
    #images (batchsize, h, w, 3)
    #cams (batchsize, 1, h, w)
    images = images.cpu().numpy()
    cams = cams.detach().cpu().numpy()
    for i in range(images.shape[0]):
        if i % 2 == 0:
            cor = '_cor'
        else:
            cor = '_not'
        cam = cams[i, 0]
        image = images[i]
        #recover image to 0-1
        image = image * np.array([0.229, 0.224, 0.225]) + np.array([0.485, 0.456, 0.406])
        image = np.clip(image, 0, 1)
        #generate heatmap
        cam = cam * 255
        cam = cv2.applyColorMap(cam.astype(np.uint8), cv2.COLORMAP_JET)
        cam = cam[:, :, ::-1] / 255
        plt.imsave('vis/img_'+str(e)+'_'+str(i)+cor+'.jpg', 0.5*cam+0.5*image)
I don't see the function called in tool.py ,and I'm so confused ,I will truly appreciate it if you can explain it for me!

**Comment by echo0409** [comment]
In the provided code, we didn’t visualize the results.  When you need to visualize some sample, you can call this function in your  own code.

**Comment by multimedia-infomation-security** [comment]
Thank you very much for your answer!I will try to call it in my code.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Yake ***@***.***&gt;; 
发送时间: 2021年12月16日(星期四) 下午3:37
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [DTaoo/Discriminative-Sounding-Objects-Localization] Some problems in tool.py (Issue #13)





 
In the provided code, we didn’t visualize the results.  When you need to visualize some sample, you can call this function in your  own code.
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by multimedia-infomation-security** [comment]
I have another question, How did you get the location visualizing results with boxes in the experiment part of your thesis?




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Yake ***@***.***&gt;; 
发送时间: 2021年12月16日(星期四) 下午3:37
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [DTaoo/Discriminative-Sounding-Objects-Localization] Some problems in tool.py (Issue #13)





 
In the provided code, we didn’t visualize the results.  When you need to visualize some sample, you can call this function in your  own code.
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by echo0409** [comment]
> 

The boxes are generated by pretrained detection model. Please read  Section 4.1, 'Bounding box annotation' of the paper for details.

**Comment by multimedia-infomation-security** [comment]
thank you again for your help！
